en/ui-system/components/engine/ui-batch.md

```diff
@@ -1,19 +1,91 @@
-- Layer of node: since layer is involved in rendering or not, different layers cannot be batched.
-- Material: the same material is a necessary requirement for batching, because the material instantiation mechanism is used, the material will be instantiated after the user sets the uniform of the material, and the material after instantiation cannot be batching. After the uniform value is used, reassign the material resource to the component to make the component participate in the batch again (through the CustomMaterial interface).
-- BlendState State: for some 2D rendering components, set the BlendState value of the component in the panel.
-- DepthStencilState state: this value controls the depth detection and stencil buffering of the component, generally the user does not need to care about the setting of this value, it is automatically controlled by the engine (for the Mask effect)
-- Texture Source and Texture Sampling: generally speaking, this condition is the most important one that affects batching, especially for sprites and text, where texture can easily make a difference and prevent batching. The engine provides some methods to achieve better batching, which can be found in the following article.
+To batch render 2D renderable components, the following criteria are required:
 
-## Batching Method Description
+| Prerequisite | Description |
+| :-- | :--- |
+| Nodes are hosted in the same layer | Renderable components in different layers are unable to be batched as layers predetermine the render process. |
+| Renderable components are drawn with the same material | Cocos Creator instantiates materials after `uniforms` are set which prevents batching. **It is prominent for renderable components to be batched to share the same material.** <br>If a `uniform` is created for a custom material and the material is to be batched after said `uniform` is no longer being used, reassign the material to the component via interface `CustomMaterial`. |
+| Material instances share the same settings for `BlendState` and `DepthStencilState` | `DepthStencilState` is automatized by the engine to perform depth testing and create stencil buffers. It is primarily used to implement visual effects such as masking. In general, it is not necessary for users to modify its parameters. |
+| Vertex data are transferred in the same **buffer** (new in v3.4.1.) | Vertex data are automatized by the engine under most scenarios and require no manual management. For more information, please refer to the **MeshBuffer Guidelines** section below. |
+| Textures share the same source and sampler type | The most common issue that prevents batching is a mismatch between textures. For example, Sprites and Labels are unable to be batched due to different sampler types. It is recommended to use a general workflow to achieve optimal batching results in Cocos Creator. Please refer to the **Batching Workflow Guidelines** section below for more details. |
```

Comment:
Sprites and Labels are unable to be batched due to the fact that labels are using independent textures.